include/nbl/builtin/hlsl/cpp_compat/impl/intrinsics_impl.hlsl

```diff
@@ -510,19 +531,57 @@ struct refract_helper<T, U>
 	}
 };
 
-#endif // C++ only specializations
+template<typename T>
+requires concepts::FloatingPoint<T>
+struct nMin_helper<T>
+{
+	using return_t = T;
+	static inline return_t __call(const T a, const T b)
+	{
+		if (std::isnan(a))
+			return b;
+		if (std::isnan(b))
+			return a;
 
-// C++ and HLSL specializations
+		return std::min(a, b);
+	}
+};
+
+template<typename T>
+requires concepts::FloatingPoint<T>
+struct nMax_helper<T>
+{
+	using return_t = T;
+	static inline return_t __call(const T a, const T b)
+	{
+		if (std::isnan(a))
+			return b;
+		if (std::isnan(b))
+			return a;
+
+		return std::max(a, b);
```

Comment:
faster way
```cpp
// comparison involving any NaN always returns false
return (b>a || isnan(a)) ? b:a;
```

Also you can't use `std::isnan` because its vulnerable to `ffast-math` you need to use our nan helper